### PR TITLE
add an option to disable telemetry, document available metrics

### PR DESCRIPTION
**Pull request by arbll**
Fixes https://github.com/DataDog/datadog-go/issues/114

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,6 +86,24 @@ After the client is created, you can start sending Service Checks to Datadog. Se
 
 ## Performance / Metric drops
 
+### Monitoring this client
+
+This client automatically injects telemetry about itself in the dogstatsd stream.
```
Comment (cswatt) [inline README.md]:
```suggestion
This client automatically injects telemetry about itself in the DogStatsD stream.
```

File: README.md [hunk 2]
```diff
@@ -86,6 +86,24 @@ After the client is created, you can start sending Service Checks to Datadog. Se
...
+- `datadog.dogstatsd.client.packets_sent` (count): total number of datagrams successfully sent
+- `datadog.dogstatsd.client.bytes_sent` (count): total number of bytes successfully sent
+- `datadog.dogstatsd.client.packets_dropped` (count): total number of datagrams dropped
+- `datadog.dogstatsd.client.bytes_dropped` (count): total number of bytes dropped
+- `datadog.dogstatsd.client.packets_dropped_queue` (count): total number of datagrams dropped because the client queue was full
+- `datadog.dogstatsd.client.bytes_dropped_queue` (count): total number of bytes dropped because the client queue was full
+- `datadog.dogstatsd.client.packets_dropped_writer` (count): total number of datagrams dropped because of an error while writing to the transport
+- `datadog.dogstatsd.client.bytes_dropped_writer` (count): total number of bytes dropped because of an error while writing to the transport
+
+Note: When using UDP datagrams may be dropped during transport. Those drops will not show up here as the client has no way to know
```
Comment (cswatt) [inline README.md]:
```suggestion
Note: When using UDP, datagrams may be dropped during transport. Those drops will not show up here as the client has no way to know.
```

File: statsd/options.go [hunk 3]
```diff
@@ -49,6 +51,9 @@ type Options struct {
 	SenderQueueSize int
 	// WriteTimeoutUDS is the timeout after which a UDS packet is dropped.
 	WriteTimeoutUDS time.Duration
+	// Telemetry is a set en metrics automatically injected by the client in the
```
Comment (wenwei63029869) [inline statsd/options.go]: It should be `Telemetry is a set of metrics`?

File: README.md [hunk 4]
```diff
@@ -86,6 +86,24 @@ After the client is created, you can start sending Service Checks to Datadog. Se
 
 ## Performance / Metric drops
 
+### Monitoring this client
+
+This client automatically injects telemetry about itself in the dogstatsd stream.
+The following metrics are available:
+
+- `datadog.dogstatsd.client.packets_sent` (count): total number of datagrams successfully sent
```
Comment (l0k0ms) [inline README.md]: Could we make a table for this list ? 

| Metric name | Type | Description | 
| --- | --- | --- |

File: statsd/options.go (hunk 3 above)
Comment (arbll) [reply statsd/options.go]: Indeed!

File: README.md (hunk 4 above)
Comment (arbll) [reply README.md]: Done!

**Pull request merged**